dashboard/Dockerfile

```diff
@@ -1,5 +1,9 @@
 FROM node:8.9 AS build
 WORKDIR /app
+
+COPY package.json yarn.lock /app/
+RUN yarn install --frozen-lockfile
```

Comment:
This improves docker cache hits in the development workflow

Reply:
what do you mean with "this"? the `--frozen-lockfile`? Why not doing the `yarn install` after the `COPY . /app`?

Reply:
> the --frozen-lockfile?

Apparently `yarn install` could go an update the yarn.lock file and update the package versions to something completely different from what it was tested against. For this reason the `--frozen-lockfile` is recommended for use in CI builds https://github.com/yarnpkg/yarn/issues/749

> Why not doing the yarn install after the COPY . /app?

Specifically w.r.t the developer use-case, this ensures that the docker layer cache gets reused during the builds. As a result image builds will be quicker when the dashboard sources are updated. Only when there are updates to the package.json/yarn.lock file, there will be a cache miss and `yarn install` will execute again.

Reply:
ah, true